pkg/k8s/apis/cilium.io/utils.go

```diff
@@ -135,11 +143,11 @@ func ParseToCiliumRule(namespace, name string, r *api.Rule) *api.Rule {
 		copy(retRule.Egress, r.Egress)
 	}
 
-	// Convert resource name to a Cilium policy rule label
-	label := fmt.Sprintf("%s=%s", PolicyLabelName, name)
-
-	// TODO: Warn about overwritten labels?
-	retRule.Labels = labels.ParseLabelArray(label)
+	policyLbls := GetPolicyLabels(namespace, name)
+	if retRule.Labels == nil {
+		retRule.Labels = make(labels.LabelArray, 0, len(policyLbls))
```

Comment:
Nitpick: can you just assign `policyLbls` to `retRule.Labels` here and put line 150 in an `else` block?

Reply:
No because it will overwrite the `retRule.Labels` that can be set by the users.